application/forms/SourceForm.php

```diff
@@ -0,0 +1,311 @@
+                'required'  => true,
+                'value'     => 'icinga',
+                'options'   => [
+                    'icinga' => 'Icinga',
+                    'other' => $this->translate('Other')
+                ]
+            ]
+        );
+
+        $chosenType = $this->getPopulatedValue('icinga_or_other');
+        $configuredType = $this->getPopulatedValue('type');
+        $showIcingaApiConfig = $chosenType === 'icinga'
+            || ($chosenType === null && $configuredType === null)
+            || ($chosenType === null && $configuredType === Source::ICINGA_TYPE_NAME);
+
+        if ($showIcingaApiConfig) {
+            // TODO: Shouldn't be necessary: https://github.com/Icinga/ipl-html/issues/131
+            $this->clearPopulatedValue('type');
+
+            $this->addElement(
+                'hidden',
+                'type',
+                [
+                    'required'  => true,
+                    'disabled'  => true,
+                    'value'     => Source::ICINGA_TYPE_NAME
+                ]
+            );
+            $this->addElement(
+                'text',
+                'icinga2_base_url',
+                [
+                    'required'  => true,
+                    'label'     => $this->translate('API URL')
+                ]
+            );
+            $this->addElement(
+                'text',
+                'icinga2_auth_user',
+                [
+                    'required'      => true,
+                    'label'         => $this->translate('API Username'),
+                    'autocomplete'  => 'off'
+                ]
+            );
+            $this->addElement(
+                'password',
+                'icinga2_auth_pass',
+                [
+                    'required'      => true,
+                    'label'         => $this->translate('API Password'),
+                    'autocomplete'  => 'new-password'
+                ]
+            );
+            $this->addElement(
+                'checkbox',
+                'icinga2_insecure_tls',
+                [
+                    'class'             => 'autosubmit',
+                    'label'             => $this->translate('Verify API Certificate'),
+                    'checkedValue'      => 'n',
+                    'uncheckedValue'    => 'y',
+                    'value'             => true
+                ]
+            );
+
+            /** @var CheckboxElement $insecureBox */
+            $insecureBox = $this->getElement('icinga2_insecure_tls');
+            if ($insecureBox->isChecked()) {
+                $this->addElement(
+                    'text',
+                    'icinga2_common_name',
+                    [
+                        'label'         => $this->translate('Common Name'),
+                        'description'   => $this->translate(
+                            'The CN of the API certificate. Only required if it differs from the host name'
```

Comment:
Maybe emphasize that this is the FQDN from the _API URL_ field above as no other "host name" was mentioned otherwise?

Reply:
What is the fqdn? The host in the url or the cn of the certificate? My understanding is, if the cn in the certificate doesn't match the actual hostname (as in the url field), one needs to define it here.

Reply:
Sorry, the Fully Qualified Domain Name of the URL from the `icinga2_base_url` field, labeled as "API URL".